### PR TITLE
Avoid writing binary to logs

### LINKED ISSUE #4130: lakeFS logs write unparsed binary data in the kv store

### PR DESCRIPTION
**Pull request by itaiad200**
closes #4130

### CODE REVIEW

File: pkg/kv/dynamodb/store.go [hunk 1]
```diff
@@ -219,7 +219,7 @@ func (s *Store) Get(ctx context.Context, partitionKey, key []byte) (*kv.ValueWit
 	var item DynKVItem
```
Comment (N-o-Z) [inline pkg/kv/dynamodb/store.go:219]: You missed one at 216

Comment (itai-david) [inline pkg/kv/dynamodb/store.go:219]: I think you forgot one on line 216
(Can't add a comment on the exact line, as it is uncahnged)

Comment (itaiad200) [reply pkg/kv/dynamodb/store.go:219]: Yikes. Thanks.

File: pkg/kv/dynamodb/store.go [hunk 2]
```diff
@@ -206,7 +206,7 @@ func (s *Store) Get(ctx context.Context, partitionKey, key []byte) (*kv.ValueWit
 	dynamoRequestDuration.WithLabelValues(operation).Observe(time.Since(start).Seconds())
 	if err != nil {
 		dynamoFailures.WithLabelValues(operation).Inc()
-		return nil, fmt.Errorf("get item: %s (key=%v): %w", err, string(key), kv.ErrOperationFailed)
+		return nil, fmt.Errorf("get item: %w", err)
```
Comment (itai-david) [inline pkg/kv/dynamodb/store.go:209]: I think this one (and all the following that converts `key` into string, actually logs the key in a pretty readable format. Are you sure we want to discard these?

**Comment by itaiad200** [comment]
> I understand where this is coming from, but I still think it might benefit from having this information. Can we maybe print it to a trace log instead?

The HTTP logging middleware prints this information. With the proper wrapping along the way, we would know what db request failed. Notice that the keys are binary, you can't assume encoding. An example of a log line:
```
"context canceled: operation failed (key=[99 100 52 97 100 100 99 55 54 51 55 48 57 100 57 100 56 98 99 49  52 56 99 53 56 50 53 100 48 101 101 48 54 56 51 101 54 97 51 49 102 97 99 48 53 55 57 111 109 109 105 116 115 47 98 55 54 49 102 50 49 50 100 100 48 100 102 54 98 49 53 54])"

```

**Review by itai-david — COMMENTED** [review]
Suggestion/Opinion : I think the postgres store is the one printing unreadable byte arrays, as it uses the `key` as is. In dynamodb store, these are converted to strings and are mostly readable. I think we should consider leaving these keys and use the same conversion for the keys in the other stores (postgres, mem), as this info might make it a lot easier to focus on any store related failure

**Review by N-o-Z — APPROVED** [review]
Lgtm

**Comment by itaiad200** [comment]
> Suggestion/Opinion : I think the postgres store is the one printing unreadable byte arrays, as it uses the `key` as is. In dynamodb store, these are converted to strings and are mostly readable. I think we should consider leaving these keys and use the same conversion for the keys in the other stores (postgres, mem), as this info might make it a lot easier to focus on any store related failure

How are they converted to strings? If you mean DynamoDB UI, then it's base64 encoded and that's not helpful to debug anything:
image

**Pull request merged**